pydough/sqlglot/transform_bindings.py

```diff
@@ -518,60 +574,259 @@ def convert_slice(
         converted to SQLGlot expressions.
 
     Returns:
-        The SQLGlot expression matching the functionality of `SUBSTRING`.
+        The SQLGlot expression matching the functionality of Python based string slicing
+        with the caveat that it only supports a step of 1.
     """
     assert len(sql_glot_args) == 4
+    _, start, stop, step = sql_glot_args
 
     for arg in sql_glot_args[1:]:
         if not isinstance(arg, (sqlglot_expressions.Literal, sqlglot_expressions.Null)):
             raise NotImplementedError(
                 "SLICE function currently only supports the slicing arguments being integer literals or absent"
             )
 
-    _, start, stop, step = sql_glot_args
-    start_idx: int | None = None
-    stop_idx: int | None = None
-    if isinstance(start, sqlglot_expressions.Literal):
-        if int(start.this) < 0:
-            raise NotImplementedError(
-                "SLICE function currently only supports non-negative start indices"
-            )
-        start_idx = int(start.this)
-    if isinstance(stop, sqlglot_expressions.Literal):
-        if int(stop.this) < 0:
-            raise NotImplementedError(
-                "SLICE function currently only supports non-negative stop indices"
-            )
-        stop_idx = int(stop.this)
     if isinstance(step, sqlglot_expressions.Literal):
-        if int(step.this) != 1:
+        try:
+            step_int = int(step.this)
+        except ValueError:
+            raise ValueError(
```

Comment:
It's better to also add which argument was the issue.
e.g. SLICE function step argument must be integer literal or absent